### PR TITLE
Fix to the issue: Column values sometimes return as Strings for non-s…

### PR DESCRIPTION
**Pull request by pamboo80**
…tring types (https://github.com/rockset/rockset-java-client/issues/7)

### CODE REVIEW

**Comment by dhruba** [comment]
i wonder if it is better to arrange these fields in RocksetTable.getColumns() https://github.com/rockset/rockset-java-client/blob/master/src/main/java/com/rockset/jdbc/RocksetTable.java#L119 instead of in the constructor @kwadhwa18 ?

**Comment by pamboo80** [comment]
100% true Dhruba. Not overloading the object initialization. In fact the describe member is used only in the getColumns() method. We can move it inside to the method before looping through. Thanks.

File: src/main/java/com/rockset/jdbc/RocksetTable.java [hunk 1]
```diff
@@ -45,7 +49,40 @@ public RocksetTable(String catalog, String schema, String tableNamePattern,
...
+        return !mapper.readTree(mapper.writeValueAsString(fieldName)).get("type").asText().equals("null");
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+      return false;
+    })
+    .collect(Collectors.toList());
+
+    //Get all null type rows
+    List<Object>fieldsDescriptionWithTypeAsNull = fieldsDescription.stream()
```
Comment (kwadhwa18) [inline src/main/java/com/rockset/jdbc/RocksetTable.java:70]: nit: space after `List<Object>`

File: src/main/java/com/rockset/jdbc/RocksetTable.java [hunk 2]
```diff
@@ -45,7 +49,40 @@ public RocksetTable(String catalog, String schema, String tableNamePattern,
...
+                return mapper.readTree(mapper.writeValueAsString(fieldName)).get("type").asText().equals("null");
+              } catch (IOException e) {
+                e.printStackTrace();
+              }
+              return false;
+            })
+            .collect(Collectors.toList());
+
+    //Combine them back
+    fieldsDescription= Stream.concat(fieldsDescriptionWithTypeAsNonNull.stream(), fieldsDescriptionWithTypeAsNull.stream())
```
Comment (kwadhwa18) [inline src/main/java/com/rockset/jdbc/RocksetTable.java:82]: nit: space after `fieldsDescription`

File: src/main/java/com/rockset/jdbc/RocksetTable.java [hunk 3]
```diff
@@ -45,7 +49,40 @@ public RocksetTable(String catalog, String schema, String tableNamePattern,
...
+    })
+    .collect(Collectors.toList());
+
+    //Get all null type rows
+    List<Object>fieldsDescriptionWithTypeAsNull = fieldsDescription.stream()
+            .filter(fieldName -> {
+              try {
+                return mapper.readTree(mapper.writeValueAsString(fieldName)).get("type").asText().equals("null");
+              } catch (IOException e) {
+                e.printStackTrace();
```
Comment (kwadhwa18) [inline src/main/java/com/rockset/jdbc/RocksetTable.java:75]: use `log` to print the stack trace. Same for above exception. Also is it fine to ignore the exception or do we need to propagate it to the user?

**Review by kwadhwa18 — COMMENTED** [review]
also as @dhruba mentioned, lets re-arrange these records in RocksetTable.getColumns()

File: src/main/java/com/rockset/jdbc/RocksetTable.java (hunk 3 above)
Comment (nagarockset) [reply src/main/java/com/rockset/jdbc/RocksetTable.java:75]: No need to propagate it to the user. The recordset already getting dumped in the old code itself. So I feel just handle the exception is fine and I will remove the printStackTrace.

**Pull request merged**